homework1/task5/sub_array/sub_array.py

```diff
@@ -0,0 +1,31 @@
+"""
+Given a list of integers numbers "nums".
+You need to find a sub-array with length less equal to "k", with maximal sum.
+The written function should return the sum of this sub-array.
+Examples:
+    nums = [1, 3, -1, -3, 5, 3, 6, 7], k = 3
+    result = 16
+"""
+from typing import List
+
+
+def find_maximal_subarray_sum(nums: List[int], k: int) -> int:
+    """
+    Function find maximal sum of sub-array with length less equal to "k"
+    :param nums: integer list
+    :param k: length of element in sub-array
+    :return: sum
+    """
+    assert k <= len(nums)
```

Comment:
k can be greater then len(nums) and its normal case because result sub-array is **less** equal to k